src/Console/WorkCommand.php

```diff
@@ -26,17 +17,23 @@ class WorkCommand extends IlluminateWorkCommand
 
     /**
      * WorkCommand constructor.
+     *
      * @param Worker $worker
+     * @param array  $config
      */
-    public function __construct(Worker $worker)
+    public function __construct(Worker $worker, $config)
     {
-        $this->renameCommandInSignature();
+        $this->renameCommandInSignature($config['command_name']);
 
         parent::__construct($worker);
     }
 
-    public function renameCommandInSignature()
+    public function renameCommandInSignature($commandName)
     {
-        $this->signature = str_replace('queue:work', $this->name, $this->signature);
+        if ($commandName) {
+            $this->signature = preg_replace(
```

Comment:
Just looking now. I dont remember why I changed this from the str_replace. Reviewing....